apps/desktop/src/app/sync/sources/dropbox.ts

```diff
@@ -39,104 +30,17 @@ class DropboxImpl implements ISourceConnector {
-    };
-    const request = query
-      ? fetch(`https://api.dropboxapi.com/2/files/${nextPage ? 'search/continue_v2' : 'search_v2'}`, {
-          method: 'POST',
-          headers: {
-            Authorization: `Bearer ${localStorage.getItem(TOKEN || '')}`,
-            'Content-Type': 'application/json',
-          },
-          body: JSON.stringify(nextPage ? { cursor: nextPage } : { query }),
-        }).then(success, failure)
-      : fetch(`https://api.dropboxapi.com/2/files/list_folder${nextPage ? '/continue' : ''}`, {
-          method: 'POST',
-          headers: {
-            Authorization: `Bearer ${localStorage.getItem(TOKEN || '')}`,
-            'Content-Type': 'application/json',
-          },
-          body: JSON.stringify(nextPage ? { cursor: nextPage } : { path: '', recursive: true, limit: pageSize }),
-        }).then(success, failure);
-    return this.authenticate().pipe(
-      filter((isSigned) => isSigned),
-      take(1),
-      switchMap(() =>
-        from(request).pipe(
-          map((result: any) => ({
-            items:
-              (query
-                ? result.matches?.filter(this.filterResults).map(this.mapResults)
-                : result.entries?.filter(this.filterFiles).map(this.mapFiles)) || [],
-            nextPage: result.has_more ? this._getFiles(query, pageSize, result.cursor) : undefined,
-          })),
-        ),
-      ),
-    );
-  }
-
-  /* eslint-disable  @typescript-eslint/no-explicit-any */
-  private mapFiles(raw: any): SyncItem {
-    return {
-      title: raw.name || '',
-      originalId: raw.id || '',
-      metadata: {},
-      status: FileStatus.PENDING,
-      uuid: raw.uuid || '',
-    };
-  }
-
-  /* eslint-disable  @typescript-eslint/no-explicit-any */
-  private mapResults(raw: any): SyncItem {
+  getParametersValues(): ConnectorParameters {
     return {
-      title: raw.metadata?.metadata?.['name'] || '',
-      originalId: raw.metadata?.metadata?.['id'] || '',
-      metadata: {},
-      status: FileStatus.PENDING,
-      uuid: raw.metadata?.metadata?.['uuid'] || '',
+      token: localStorage.getItem(TOKEN),
     };
   }
 
-  /* eslint-disable  @typescript-eslint/no-explicit-any */
-  private filterFiles(raw: any): boolean {
-    return raw?.['.tag'] !== 'folder';
-  }
-
-  /* eslint-disable  @typescript-eslint/no-explicit-any */
-  private filterResults(raw: any): boolean {
-    return raw.match_type?.['.tag'] !== 'folder';
+  goToOAuth(reset?: boolean) {
+    return of(true);
   }
 
-  download(resource: SyncItem): Observable<Blob> {
-    return from(
-      fetch(`https://content.dropboxapi.com/2/files/download`, {
-        method: 'POST',
-        headers: {
-          Authorization: `Bearer ${localStorage.getItem(TOKEN || '')}`,
-          'Dropbox-API-Arg': JSON.stringify({ path: resource.originalId }),
-        },
-      }).then((res) => res.blob()),
-    );
+  authenticate(): Observable<boolean> {
+    return of(!!this.getParametersValues().token);
```

Comment:
seems like this method is checking if the connector is authenticated, so the name `authenticate` is misleading

Reply:
Here again some connectors need to run some code to authenticate, this is where it is supposed to happen, but for DropBox there is nothing to do, so as far as the token is defined, we can continue.